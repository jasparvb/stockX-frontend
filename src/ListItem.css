.ListItem {
  display: flex;
  justify-content: space-around;
}