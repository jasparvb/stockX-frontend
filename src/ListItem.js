import React from "react";

function ListItem({id, ticker, name, listId, removeStock}) {

  return (
    <div className="List">
      <h4>{ticker}</h4>
      <p>{name}</p>
      <button onClick={() => removeStock(id, listId)}>X</button>
    </div>
  );
}

export default ListItem;