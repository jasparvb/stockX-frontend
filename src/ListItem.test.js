import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from "react-redux";
import rootReducer from "./reducers/root";
import { createStore } from "redux";
import { BrowserRouter } from "react-router-dom";
import ListItem from "./ListItem";

const store = createStore(rootReducer);

const quote = {
  open: 114.68,
  close: 112,
  change: -1.49,
  changePercent: -0.01313
}

it("renders without crashing", function() {
  render(
    <Provider store={store}>
      <BrowserRouter>
        <ListItem quote={quote} />
      </BrowserRouter>
    </Provider>);
});


it("matches snapshot", function() {
  const { asFragment } = render(
    <Provider store={store}>
      <BrowserRouter>
        <ListItem quote={quote} />
      </BrowserRouter>
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});
