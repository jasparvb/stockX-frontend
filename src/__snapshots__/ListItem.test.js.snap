// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ListItem px-5"
  >
    <a
      class="text-left col-md-5"
      href="/undefined"
    >
      <h4
        class="mb-0"
      />
      <p />
    </a>
    <div
      class="text-right col-md-5"
    >
      <h4
        class="mb-0"
      >
        $112.00
      </h4>
      <p
        class="percent-change down"
      >
        -1.31%
      </p>
    </div>
    <button
      class="btn remove"
    >
      <i
        class="fas fa-times"
      />
    </button>
  </div>
</DocumentFragment>
`;
